### PR TITLE
Remove NuMTs from MT pipeline and updates wdl to GATK4.1.1.0

### PR DESCRIPTION
**Pull request by meganshand**
Tests are not passing because I'm now using NIO in the WDL. I'll need to fix that, but the WDL itself should be ready for review. 

The changes:
- Updates the pipeline for the new Mutect2 Filtering scheme and pulls filtering after the liftover and recombining of the VCF. 
- Makes the subsetting of the WGS bam fast by using PrintReads over just chrM instead of traversing the whole bam for NuMT mates.
- Moves polymorphic NuMTs based on autosomal coverage to a filter (it was an annotation before)
- Adds option to hard filter by VAF

### CODE REVIEW

File: scripts/mitochondria_m2_wdl/AlignAndCall.wdl [hunk 1]
```diff
@@ -335,3 +355,172 @@ task LiftoverAndCombineVcfs {
...
+  Float? f_score_beta
+
+  File blacklisted_sites
+  File blacklisted_sites_index
+
+  File? gatk_override
+
+  # runtime
+  Int? preemptible_tries
+  Int disk_size = 50
```
Comment (jsotobroad) [inline scripts/mitochondria_m2_wdl/AlignAndCall.wdl]: hard coded disk?

File: scripts/mitochondria_m2_wdl/AlignAndCall.wdl [hunk 2]
```diff
@@ -335,3 +355,172 @@ task LiftoverAndCombineVcfs {
...
+  runtime {
+      docker: "us.gcr.io/broad-gatk/gatk:4.1.1.0"
+      memory: machine_mem + " MB"
+      disks: "local-disk " + disk_size + " HDD"
+      preemptible: select_first([preemptible_tries, 5])
+      cpu: 2
+  }
+  output {
+      File raw_vcf = "${output_vcf}"
+      File faw_vcf_idx = "${output_vcf_index}"
```
Comment (jsotobroad) [inline scripts/mitochondria_m2_wdl/AlignAndCall.wdl]: is that supposed to be `raw_vcf_idx`?

File: scripts/mitochondria_m2_wdl/AlignAndCall.wdl [hunk 3]
```diff
@@ -335,3 +355,172 @@ task LiftoverAndCombineVcfs {
...
+  File ref_fasta
+  File ref_fai
+  File ref_dict
+  File raw_vcf
+  File raw_vcf_index
+  File raw_vcf_stats
+  Boolean compress
+  Float? vaf_cutoff
+
+  String output_vcf = "output" + if compress then ".vcf.gz" else ".vcf"
```
Comment (jsotobroad) [inline scripts/mitochondria_m2_wdl/AlignAndCall.wdl]: do we really need to allow the pipeline to output uncompressed vcfs?

Comment (meganshand) [reply scripts/mitochondria_m2_wdl/AlignAndCall.wdl]: I only look at uncompressed vcfs (the files are already tiny and it makes looking at things in Terra/Notebooks smoother). In fact I think I hardcoded `compress = false` when calling this task. Should I wire that argument through as an option?

File: scripts/mitochondria_m2_wdl/AlignAndCall.wdl [hunk 4]
```diff
@@ -335,3 +355,172 @@ task LiftoverAndCombineVcfs {
...
+      touch bamout.bam
+
+      gatk --java-options "-Xmx2500m" FilterMutectCalls -V ${raw_vcf} \
+        -R ${ref_fasta} \
+        -O filtered.vcf \
+        --stats ${raw_vcf_stats} \
+        ${m2_extra_filtering_args} \
+        --max-alt-allele-count ${max_alt_allele_count} \
+        --mitochondria-mode \
+        ${"--autosomal-coverage " + autosomal_coverage} \
```
[the remark this answers is not in the document]
Comment (meganshand) [reply scripts/mitochondria_m2_wdl/AlignAndCall.wdl]: My thought was that if a user doesn't have this information it should be optional. But if you don't provide an input here you don't get the polymorphic NuMT filter, so I'm open to making it required. Do you have a strong opinion?

File: scripts/mitochondria_m2_wdl/MitochondriaPipeline.wdl [hunk 5]
```diff
@@ -9,15 +9,12 @@ workflow MitochondriaPipeline {
     wgs_aligned_input_bam_or_cram: "Full WGS hg38 bam or cram"
     autosomal_coverage: "Median coverage of full input bam"
     out_vcf: "Final VCF of mitochondrial SNPs and INDELs"
+    vaf_filter_threshold: "Hard threshold for filtering low VAF sites"
+    f_score_beta: "F-Score beta balances the filtering strategy between recall and precision. The relative weight of recall to precision."
   }
   File wgs_aligned_input_bam_or_cram
-  Int? autosomal_coverage
+  Float? autosomal_coverage
```
[the remark this answers is not in the document]
Comment (meganshand) [reply scripts/mitochondria_m2_wdl/MitochondriaPipeline.wdl]: Do you mean a file with only that float in it? Or do you mean Picard's coverage metrics file? And if it's the latter, would I have to put the code that reads the file and extracts that value directly in GATK or in the WDL?

File: scripts/mitochondria_m2_wdl/MitochondriaPipeline.wdl [hunk 6]
```diff
@@ -156,76 +150,43 @@ workflow MitochondriaPipeline {
...
-  }
   command <<<
     set -e
-
     export GATK_LOCAL_JAR=${default="/root/gatk.jar" gatk_override}
 
-    gatk --java-options "-Xmx2500m" AddOriginalAlignmentTags \
+    gatk PrintReads \
+      ${"-R " + ref_fasta} \
+      -L chrM \
```
[the remark this answers is not in the document]
Comment (meganshand) [reply scripts/mitochondria_m2_wdl/MitochondriaPipeline.wdl]: I put it as an input to the workflow: `String contig_name = "chrM"` which my understanding from the spec means that a user can overwrite this value.

File: scripts/mitochondria_m2_wdl/AlignAndCall.wdl (hunk 4 above)
[the remark this answers is not in the document]
Comment (jsotobroad) [reply scripts/mitochondria_m2_wdl/AlignAndCall.wdl]: gotcha that makes sense to me, also your `parameter_meta` 👯 is great!

File: scripts/mitochondria_m2_wdl/MitochondriaPipeline.wdl (hunk 5 above)
[the remark this answers is not in the document]
Comment (jsotobroad) [reply scripts/mitochondria_m2_wdl/MitochondriaPipeline.wdl]: I was thinking about Picard's metrics file in the case where a non data delivery person wants to run from start to end (they don't get the coverage automatically put into their data model).  It's probably not worth the trouble at this point though.

File: scripts/mitochondria_m2_wdl/AlignAndCall.wdl (hunk 3 above)
Comment (jsotobroad) [reply scripts/mitochondria_m2_wdl/AlignAndCall.wdl]: Ehhh it just irks me when I see uncompressed output (even if I know its tiny).  Tools you use in Terra/Notebooks should be able to read compressed vcfs :P.  I would lean towards wiring it through so people have an option to changing it.

File: scripts/mitochondria_m2_wdl/AlignAndCall.wdl [hunk 7]
```diff
@@ -99,64 +98,85 @@ workflow AlignAndCall {
...
+  call MergeStats {
+    input:
+      shifted_stats = CallShiftedMt.stats,
+      non_shifted_stats = CallMt.stats,
+      gatk_override = gatk_override,
+      preemptible_tries = preemptible_tries
+  }
+
+  call Filter {
+  	input:
```
Comment (jsotobroad) [inline scripts/mitochondria_m2_wdl/AlignAndCall.wdl]: too many spaces before `input:`

File: scripts/mitochondria_m2_wdl/MitochondriaPipeline.wdl (hunk 5 above)
[the remark this answers is not in the document]
Comment (meganshand) [reply scripts/mitochondria_m2_wdl/MitochondriaPipeline.wdl]: Yeah you're right, let's leave that as future work.

**Comment by yzhang88** [comment]
Could I find any further details about how to filter the NuMTs somewhere?

**Comment by meganshand** [comment]
@yzhang88 I'd recommend looking on the GATK forum. There's been some discussion in the comments on this here: https://gatkforums.broadinstitute.org/gatk/discussion/comment/61646
Or you can search the rest of the GATK forum/open a new thread as needed. Thanks!

**Comment by yzhang88** [comment]
Thank you!

**Pull request merged**